flink-libraries/flink-table/src/main/scala/org/apache/flink/table/catalog/ExternalCatalogTable.scala

```diff
@@ -37,25 +36,10 @@ import org.apache.flink.table.plan.stats.TableStats
   * @param lastAccessTime       Timestamp of last access of the table
   */
 case class ExternalCatalogTable(
-    identifier: TableIdentifier,
```

Comment:
The properties field is exactly meant to hold configuration parameters such as hostnames, ports, paths, or client configs. 

I think @haohui is right. Not all external tables require a name to access the data (the name under which the table is registered in the catalog and identified by Calcite is the key in the map). A CSV file table needs a path, a Kafka-based table a Kafka config + topic name, etc. A database table which needs a table name could store the table name (and also the database name and schema name) in the properties map.

Would that work for you @KurtYoung?